virtual_rainforest/models/plants/plants_model.py

```diff
@@ -0,0 +1,147 @@
+        constants: Set of constants for the plants model.
+    """
+
+    model_name = "plants"
+    """An internal name used to register the model and schema"""
+    lower_bound_on_time_scale = "1 day"
+    """Shortest time scale that plants model can sensibly capture."""
+    upper_bound_on_time_scale = "1 year"
+    """Longest time scale that plants model can sensibly capture."""
+    required_init_vars = (
+        ("plant_cohorts_n", tuple()),
+        ("plant_cohorts_pft", tuple()),
+        ("plant_cohorts_cell_id", tuple()),
+        ("plant_cohorts_dbh", tuple()),
+    )
+    """Required initialisation variables for the plants model.
+
+    This is the set of variables and their core axes that are required in the data
+    object to create a PlantsModel instance."""
+
+    # TODO - think about a shared "plant cohort" core axis that defines these, but the
+    #        issue here is that the length of this is variable.
+
+    vars_updated = (
+        "leaf_area_index",  # NOTE - LAI is integrated into the full layer roles
+        "layer_heights",  # NOTE - includes soil, canopy and above canopy heights
+        "herbivory",
+    )
+    """Variables updated by the plants model."""
+
+    def __init__(
+        self,
+        data: Data,
+        update_interval: Quantity,
+        plant_functional_types: PlantFunctionalTypes,
+        constants: PlantsConsts,
+        **kwargs: Any,
+    ):
+        super().__init__(data, update_interval, **kwargs)
+
+        # Save the class attributes
+        self.pfts = plant_functional_types
+        """The plant functional types used in the plants model."""
+        self.constants = constants
+        """Set of constants for the plants model"""
+
+        self.communities = PlantCommunities(data, self.pfts)
+        """Initialise the plant communities from the data object."""
+
+        # TODO - initialise the canopy model
+
+    @classmethod
+    def from_config(
+        cls, data: Data, config: dict[str, Any], update_interval: Quantity
+    ) -> PlantsModel:
+        """Factory function to initialise a plants model from configuration.
+
+        This function returns a PlantsModel instance based on the provided configuration
+        and data, raising an exception if the configuration is invalid.
+
+        Args:
+            data: A :class:`~virtual_rainforest.core.data.Data` instance.
+            config: A validated :class:`~virtual_rainforest.core.config.Config`
+                instance.
+            update_interval: Frequency with which all models are updated
+        """
+
+        # Check if any constants have been supplied
+        if "plants" in config and "constants" in config["plants"]:
+            # Checks that constants is config are as expected
+            check_valid_constant_names(config, "plants", "PlantsConsts")
+            # If an error isn't raised then generate the dataclass
+            constants = PlantsConsts(**config["soil"]["constants"]["SoilConsts"])
```

Comment:
```suggestion
            constants = PlantsConsts(**config["plants"]["constants"]["PlantsConsts"])
```

I assume this is what you mean here, or do you initialize any constants from the soil model?